src/main/splash.ts

```diff
@@ -8,11 +8,28 @@ import { BrowserWindow } from "electron";
 import { join } from "path";
 import { SplashProps } from "shared/browserWinProperties";
 import { VIEW_DIR } from "shared/paths";
+import tinycolor from "tinycolor2";
+
+import { Settings } from "./settings";
 
 export function createSplashWindow() {
     const splash = new BrowserWindow(SplashProps);
 
     splash.loadFile(join(VIEW_DIR, "splash.html"));
 
+    const { splashBackground, splashColor, splashTheming } = Settings.store;
+    if (splashTheming) {
+        if (splashColor) {
+            splash.webContents.insertCSS(`body { --fg: ${splashColor} !important }`);
+            const semiTrans = tinycolor(splashColor);
```

Comment:
this adds a ~30kb library for such a small thing

It's also not necessary. You can obtain parsed css colours via getComputedStyle. For example for background-primary, `getComputedStyle(document.querySelector("[class|=chatContent]")).backgroundColor` gives you it in rgba format

Reply:
It seems that all the classes that use `--text-normal` are hashed, and you can't really rely on the elements that use these classes to be available on load anyways